### PR TITLE
Add new crypto package

### LINKED ISSUE #3994: Add an enum support to hash functions
Currently we have to provide algorithm name as a string and no documentations available to provide the list of available algorithms. We need to add enum support for following functions.

`getHash`
`getHmac`
`getHmacFromBase64`

### LINKED ISSUE #4090: Adding new package for security/cryptography related functions
At the moment, hashing related functions are defined as util functions. But there is a requirement to validate JWT tokens in Ballerina (https://github.com/ballerinalang/ballerina/issues/4056) as well. So it is better if we can move these hashing related functions to a new package so all other functions related to security and cryptography can be moved to this.

### PR DESCRIPTION
**Pull request by Shan1024**
Resolves #4090 
Resolves #3994

### CODE REVIEW

File: modules/ballerina-builtin/src/main/ballerina/ballerina/security/natives.bal [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package ballerina.security;
+
+@Description { value:"Returns a hash of a given string using the SHA-256 algorithm "}
+@Param { value:"baseString: The string to be hashed" }
+@Param { value:"algorithm: The hashing algorithm to be used" }
+@Return { value:"The hashed string" }
+public native function getHash (string baseString, string algorithm) (string);
```
Comment (lankavitharana) [inline modules/ballerina-builtin/src/main/ballerina/ballerina/security/natives.bal]: Is this security related function?

**Comment by lankavitharana** [comment]
I feel some of these functions are not security related, so shall we first distinguish which is which and clean up the API?

File: modules/ballerina-builtin/src/main/ballerina/ballerina/security/natives.bal (hunk 1 above)
Comment (Shan1024) [reply modules/ballerina-builtin/src/main/ballerina/ballerina/security/natives.bal]: This is used to generate SHA1, SHA256, MD5 hashes which are used for security purposes. But we can move them to a separate package(crypro package) as well.

File: modules/ballerina-builtin/src/main/ballerina/ballerina/security/crypto/natives.bal [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package ballerina.security.crypto;
+
+@Description {value:"Returns a hash of a given string using the specified algorithm (supported algorithms: SHA1, SHA256, MD5)."}
+@Param {value:"baseString: The string to be hashed"}
+@Param {value:"algorithm: The hashing algorithm to be used"}
+@Return {value:"The hashed string"}
+public native function getHash (string baseString, string algorithm) (string);
```
Comment (anupama-pathirage) [inline modules/ballerina-builtin/src/main/ballerina/ballerina/security/crypto/natives.bal]: Shall we make this algorithm  an enum in crypto package? Having it as a string is more prone to typos and user has to refer the docs to find the supported algorithms

File: modules/ballerina-builtin/src/main/ballerina/ballerina/security/crypto/natives.bal [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+@Description {value:"Returns an HMAC value for the provided base string (supported algorithms: SHA1, SHA256, MD5)."}
+@Param {value:"baseString: The string to be hashed"}
+@Param {value:"keyString: The key string "}
+@Param {value:"algorithm: The hashing algorithm to be used"}
+@Return {value:"The hashed string"}
+public native function getHmac (string baseString, string keyString, string algorithm) (string);
```
Comment (anupama-pathirage) [inline modules/ballerina-builtin/src/main/ballerina/ballerina/security/crypto/natives.bal]: Same as above. Shall we make this algorithm an enum type
